src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        size = 0;
+        first = null;
+        last = null;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode;
+        if (size == 0) {
+            newNode = new Node<>(value, null, null);
+            first = newNode;
+        } else {
+            newNode = new Node<>(value, last, null);
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> newNode;
+        if (index == 0) {
+            newNode = new Node<>(value, null, first);
+            first.prev = newNode;
+            first = newNode;
+        } else {
+            Node<T> currentPlace = iterator(index);
+            newNode = new Node<>(value, currentPlace.prev, currentPlace);
+            currentPlace.prev.next = newNode;
+            currentPlace.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (index == size - 1) {
+            return last.element;
+        }
```

Comment:
this check is redundant, your code should work without it

Reply:
done